src/Pagination.ts

```diff
@@ -0,0 +1,104 @@
+import { ItemProperties } from './interfaces';
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import PageLink, { PageLinkProperties } from './pagination/PageLink';
+
+import * as css from './styles/pagination.m.css';
+
+export const PaginationBase = ThemeableMixin(WidgetBase);
+
+export interface PaginationProperties extends ThemeableProperties {
+	page: number;
+	pages: number;
+	statusMessage: string;
+	onPageRequest?: (pageNumber: number) => void;
+}
+
+@theme(css)
+class Pagination extends PaginationBase<PaginationProperties> {
+	render (): DNode {
+		const {
+			onPageRequest,
+			page,
+			pages,
+			statusMessage
+		} = this.properties;
+		const isFirstPage = page === 1;
+		const isLastPage = page === pages;
+		const children = [
+			v('div', {
+				classes: this.classes(css.status)
+			}, [
+				statusMessage
+			])
+		];
+		const onclick = onPageRequest ? { onPageRequest } : {};
+
+		if (pages > 1) {
+			const pageLinks = [];
+
+			pageLinks.push(w(PageLink, <PageLinkProperties> { key: '1', disabled: isFirstPage, page: 1, ...onclick }));
+
+			if (page > 4) {
+				pageLinks.push(v('span', { classes: this.classes(css.pageSkip) }, [ '...' ]));
```

Comment:
I'm pretty sure this will need a key, especially when there are two ... siblings where they'd each need separate keys because they are otherwise identical.